composition-js/src/validate.ts

```diff
@@ -401,32 +402,36 @@ export class ValidationContext {
         return !args.external && !args.usedOverridden;
       }).length > 1);
   }
-
 }
 
 export class ValidationState {
   constructor(
     // Path in the supergraph corresponding to the current state.
     public readonly supergraphPath: RootPath<Transition>,
     // All the possible paths we could be in the subgraph.
-    public readonly subgraphPaths: TransitionPathWithLazyIndirectPaths<RootVertex>[]
+    public readonly subgraphPaths: TransitionPathWithLazyIndirectPaths<RootVertex>[],
+    // When we encounter an `@override`n field with a label condition, we record
+    // its value (T/F) as we traverse the graph. This allows us to ignore paths
+    // that can never be taken by the query planner (i.e. a path where the
+    // condition is T in one case and F in another).
+    public selectedOverrideConditions: Record<string, boolean> = {},
```

Comment:
If we're using JS objects here instead of `Map`s, we should use `Object.create(null)` instead of `{}`.

Reply:
Switched to map in [dbb26c1](https://github.com/apollographql/federation/pull/2902/commits/dbb26c1b11cf6c3f10490c83121a0df9504078c4)